### DISCUSSION TITLE
Porting tokenizer to javascript

### DISCUSSION THREAD

**Discussion by mustafa0x** [CAMeL-Lab/camel_tools#126] [Q&A] [answered]
I'm interesting in using the tokenizer in javascript. I plan to tokenize user queries in the browser. How feasible is this? What would the general approach be?

Thanks!

**Comment by owo** [comment, marked as answer]
Which tokenizer specifically ([`simple_word_tokenize`](https://camel-tools.readthedocs.io/en/latest/api/tokenizers/word.html#camel_tools.tokenizers.word.simple_word_tokenize) or [`MorphologicalTokenizer`](https://camel-tools.readthedocs.io/en/latest/api/tokenizers/morphological.html#camel_tools.tokenizers.morphological.MorphologicalTokenizer))?

Unfortunately, `simple_word_tokenize` should be simple to implement in JS while `MorphologicalTokenizer` might be a lot more difficult depending on the type of disambiguator used.

At the moment, a JS implementation of CAMeL Tools is not something we plan on working on.

My suggestion is to use something like [Flask](https://flask.palletsprojects.com/en/3.0.x/) to create a web API that you can call in your web app.

**Reply by mustafa0x** [reply]
Thank you Ossama. I likely will port it at some point due to my requirements. If you could provide a simple overview of the components that need to be ported I'd appreciate that. I also understand if you're short on time!

Edit: "port it" = `MorphologicalTokenizer`
